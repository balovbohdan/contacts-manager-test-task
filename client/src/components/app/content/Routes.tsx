import * as React from 'react';
import {Route} from 'react-router';

import {Route as TRoute, routes} from '@routes';

export const Routes = () => {
    const mapper = ({path, exact, component}:TRoute) =>
        <Route key={path} path={path} exact={!!exact} component={component}/>;

    const items = routes.map(mapper);

    return <>{items}</>;
};