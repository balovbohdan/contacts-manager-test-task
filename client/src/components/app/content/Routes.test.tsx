import {shallow} from 'enzyme';
import * as React from 'react';

import {Routes} from './Routes';

describe('Routes', () => {
    it('Should match snapshot', () => {
        const routes = shallow(<Routes/>);

        expect(routes).toMatchSnapshot();
    });
});