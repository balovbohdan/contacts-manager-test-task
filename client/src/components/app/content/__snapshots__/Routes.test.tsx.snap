// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Should match snapshot 1`] = `
<Fragment>
  <Route
    component={[Function]}
    exact={true}
    key="/"
    path="/"
  />
  <Route
    component={[Function]}
    exact={false}
    key="/contacts"
    path="/contacts"
  />
</Fragment>
`;
